### PR TITLE
[RFC] Improvement #3

### PR DESCRIPTION
**Pull request by otyshchenko1**
Please note, that improvement requires corresponding support in DDK.

Currently, it consists of three patches, but I am ok to squash them all into a single commit.

### CODE REVIEW

File: tools/libxl/libxl_arm.c [hunk 1]
```diff
@@ -29,6 +29,7 @@ int libxl__arch_domain_prepare_config(libxl__gc *gc,
     unsigned int i;
     uint32_t vuart_irq;
     bool vuart_enabled = false;
+    libxl_device_vgsx *vgsx;
```
Comment (andr2000) [inline tools/libxl/libxl_arm.c]: if you want this helper variable then put it down into
`if (d_config->num_vgsxs) {`
where it is only used

File: xen/arch/arm/irq.c [hunk 2]
```diff
@@ -32,19 +34,43 @@ const unsigned int nr_irqs = NR_IRQS;
 static unsigned int local_irqs_type[NR_LOCAL_IRQS];
 static DEFINE_SPINLOCK(local_irqs_type_lock);
 
+#define GSX_MAX_OS_CNT                                     8
```
Comment (andr2000) [inline xen/arch/arm/irq.c]: this is the second time we define this constant...

File: tools/libxl/libxl_arm.c (hunk 1 above)
Comment (otyshchenko1) [reply tools/libxl/libxl_arm.c]: ok

File: xen/arch/arm/irq.c (hunk 2 above)
Comment (otyshchenko1) [reply xen/arch/arm/irq.c]: Yes, you are right (xen + libxl). For keeping it in a single place, we need to define it somewhere in a public header. Shall I do that?

**Comment by andr2000** [comment]
Does this need an update or it is still valid?

**Comment by otyshchenko1** [comment]
> Does this need an update or it is still valid?
I will update PR in a few hours.

**Comment by otyshchenko1** [comment]
Updated PR (added one more patch)

**Pull request closed without merging**